view/theme/frio/js/theme.js

```diff
@@ -764,12 +764,106 @@ function htmlToText(htmlString) {
  * @param {boolean} un    Whether to perform an activity removal instead of creation
  */
 function doActivityItemAction(ident, verb, un) {
-	if (verb.indexOf("attend") === 0) {
-		$(".item-" + ident + " .button-event:not(#" + verb + "-" + ident + ")").removeClass("active");
+	_verb = un ? 'un' + verb : verb;
+	var thumbsClass = '';
+	switch (verb) {
+		case 'like':
+			thumbsClass = 'fa-thumbs-up';
+			break;
+		case 'dislike':
+			thumbsClass = 'fa-thumbs-down';
+			break;
+		case 'announce':
+			thumbsClass = 'fa-retweet';
+	}
+	if (verb.indexOf('announce') === 0 ) {
+		// Share-Button(s)
+		// remove share-symbol, to replace it by rotator
+		$('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child').removeClass('fa-share');
+		$('button[id^=announce-' + ident.toString() + '] i:first-child').removeClass('fa-retweet');
+		// if no wait-rotator for activity(verb) is added, add it. or just show it, if exists
+		if ($('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').length == 0) {
+			// append rotator to the shareMenu-button for small media
+      $('<img>')
+        .attr('id', 'waitfor-' + verb + '-' + ident.toString())
+				.addClass('fa')
+				.appendTo($('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child' )).show();
+		} else {
+			$('<img>').attr({id: 'waitfor-' + verb + '-' + ident.toString(), src: 'images/rotator.gif'})
+		}
+	}
+	$('button[id^=' + verb + '-' + ident.toString() + '] i:first-child').removeClass(thumbsClass);
+	// if verb is announce, then one rotator is added above to the shareMedia-dropdown button
+	if ($('button:not(button.dropdown-toggle) img#waitfor-' + verb + '-' + ident.toString()).length == 0) {
+		$('<img>')
+      .attr('id', 'waitfor-' + verb + '-' + ident.toString())
+			.attr('src', 'images/rotator.gif')
+			.addClass('fa')
+			.appendTo($('button[id^=' + verb + '-' + ident.toString() + '] i:first-child')).show();
+	} else {
+		// show existing rotator for activity
+		$('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').show()
 	}
-	$("#" + verb + "-" + ident).toggleClass("active");
 
-	doActivityItem(ident, verb, un);
+	// do request for activity
+	$.post('item/' + ident.toString() + '/activity/' + _verb)
+	.success(function(data){
+		//$('#like-rotator-' + ident.toString()).hide();
+		$('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').remove();
+		if (data.status == 'ok') {
+			// response from server was ok
+			if (data.verb == 'un' + verb) {
+				// like/dislike buttons
+				$('button[id^=' + verb + '-' + ident.toString() + ']' )
+					.removeClass('active')
+					.attr('onclick', 'doActivityItemAction(' + ident +', "' + verb + '",false )').change();
```

Comment:
What's the `change()` call doing in this case?
```suggestion
					.attr('onclick', 'doActivityItemAction(' + ident +', "' + verb + '",false )')
					.change();
```